File: src/code/StraightLineFormula.java
package code;

import api.GeoLocation;

public class StraightLineFormula {
    public double slope;
    public double b_of_formula;

    public StraightLineFormula(GeoLocation first, GeoLocation second) {
        double firstX = first.x();
        double firstY = first.y();
        double secondX = second.x();
        double secondY = second.y();

        slope = (secondY - firstY)/(secondX - firstX);
        b_of_formula = firstY-(firstX*slope);
    }

    public boolean checkOnEdge(GeoLocation pos) {
        double y = pos.y();
        double toCheck = pos.x()*this.slope + this.b_of_formula;
        return (y == toCheck);
    }

    @Override
    public String toString() {
       return "y = " + slope + "*x " +b_of_formula;
    }
}

File: implementation/StraightLineFormula.py
class StraightLineFormula:
    def __init__(self):
        self.slope = 0
        self.b_of_formula = 0

    def StraightLineFormula(self, first, second):
        firstX = first[0]
        firstY = first[1]
        secondX = second[0]
        secondY = second[1]

        self.slope = (secondY - firstY) / (secondX - firstX)
        self.b_of_formula = firstY - (firstX * self.slope)

    def onEdge(self, pos):
        y = pos[1]
        y = float("{:.12f}".format(y))
        check = pos[0] * self.slope + self.b_of_formula
        check = float("{:.12f}".format(check))
        if y == check:
            return True
        return False

    def __repr__(self):
        return "y = " + self.slope + " * x" + self.b_of_formula
